### DISCUSSION TITLE
Upgrading from v2.11.7 to v3.0.0

### DISCUSSION THREAD

**Discussion by aquelf** [netbox-community/netbox#7095] [Q&A] [answered]
Hi all, 

I was looking to do my 1st upgrade, my current version is v2.11.7 so based on the docx I don't need to upgrade to an intermediate before going to v3.0.0?
Based on the docx I need to upgrade the existing x.y.z version but some of the folders or files don't exist. 

The following is the list of steps in which I'm following:
Step 1:
wget https://github.com/netbox-community/netbox/archive/v3.0.0.tar.gz
sudo tar -xzf v3.0.0.tar.gz -C /opt
sudo ln -sfn /opt/netbox-3.0.0/ /opt/netbox

Step 2:
sudo cp /opt/netbox-2.11.7/local_requirements.txt /opt/netbox/
sudo cp /opt/netbox-2.11.7/netbox/netbox/configuration.py /opt/netbox/netbox/netbox/
sudo cp /opt/netbox-2.11.7/netbox/netbox/ldap_config.py /opt/netbox/netbox/netbox/

Step 3:
sudo cp -pr /opt/netbox-2.11.7/netbox/media/ /opt/netbox/netbox/

Step 4:
sudo cp -r /opt/netbox-2.11.7/netbox/scripts /opt/netbox/netbox/
sudo cp -r /opt/netbox-2.11.7/netbox/reports /opt/netbox/netbox/

Step 5:
sudo cp /opt/netbox-2.11.7/gunicorn.py /opt/netbox/

Step 6:Run upgrade script
sudo ./upgrade.sh

Step 7: restart services
sudo systemctl restart netbox netbox-rq

Any help or documentation would be very much appreciated.

Thanks in advance.

**Comment by candlerb** [comment, marked as answer]
That looks OK to me, assuming you installed Netbox initially from a tarball rather than a git clone.  You won't need to upgrade to an intermediate version, but if your system currently has python 3.6 then you'll need to upgrade to python 3.7 or later.

Are you seeing any errors?  If so, can you show at what step and the exact error you see?

**Reply by aquelf** [reply]
TBH I haven't tried with this particular version so was just preempting in case I go down a rabbit hole.

I did try to upgrade a test ver (v2.10.8) but it seems to be the same. In both cases I don't see any folder with the version number in the opt folder /opt/netbox-2.11.7/ & don't seem to have any of the following files.

local_requirements.txt
configuration.py
ldap_config.py

I installed using the following instructions:
https://netbox.readthedocs.io/en/stable/installation/

Thanks

**Reply by candlerb** [reply]
1. When your v2.11.7 was originally installed and you were following the instructions at https://netbox.readthedocs.io/en/stable/installation/3-netbox/ did you use option A (install from git) or option B (install from tarball)?  With option A, you won't see the version number in the path, it will just be `/opt/netbox`.  And it that case you need to do the git-based upgrade (i.e. `git pull` not downloading/untarring a new release)
2. You won't have `local_requirements.txt` unless you created it.  It's for installing optional python dependencies (e.g. napalm libraries).  If you don't need them then you won't have it.
3. You won't have `configuration.py` unless you've created it.  For a brand new installation you would copy configuration.example.py to configuration.py, then edit it.  But your existing Netbox v2.11.7 installation, if it's working, will most definitely have a configuration.py (otherwise it couldn't possibly work)
4. `ldap_config.py` is a file you'd create if you're using LDAP authentication.  if you're not, then you have no need to create this.

**Reply by aquelf** [reply]
Hi, yea that makes perfect sense. I'll give that a try. Thanks a million for the prompt reply.
